version all the files, update CI tests and vscode settings

dave-connors-3 opened the pull request:
this PR 

- adds versioned schema files for 1.5 + 1.6
- leaves the originals in place until we get the greenlight to deprecate them from @benmosher 
- updates CI and the vscode settings to point the right versions at separate test files in the tests directory. 
- adds `dependencies.yml` support in 1.6

benmosher commented:
an idea: storing files as e.g. `schemas/{version}/dbt_project-{version}.json` would be the teensiest bit redundant, but as the versions pile up, you won't end up with one super-heavy folder fulla schemata.

joellabes commented on .github/workflows/ci.yml:
```diff
@@ -26,30 +25,52 @@ jobs:
+          npx ajv-cli test -s schemas/dbt_yml_files-1.5.json -d tests/1.5/valid/schema.yml --valid
+          npx ajv-cli test -s schemas/dbt_yml_files-1.5.json -d tests/1.5/invalid/schema.yml --invalid
+      - name: Validate packages.yml sample files for dbt version 1.5
+        run: |
+          npx ajv-cli test -s schemas/packages-1.5.json -d tests/1.5/valid/packages.yml --valid
+          npx ajv-cli test -s schemas/packages-1.5.json -d tests/1.5/invalid/packages.yml --invalid
+      - name: Validate selectors.yml sample files for dbt version 1.5
         run: |
-          npx ajv-cli test -s schemas/dbt_project.json -d tests/valid/dbt_project.yml --valid
-          npx ajv-cli test -s schemas/dbt_project.json -d tests/invalid/dbt_project.yml --invalid
-      - name: Validate resource yml sample files
+          npx ajv-cli test -s schemas/selectors-1.5.json -d tests/1.5/valid/selectors.yml --valid
+          npx ajv-cli test -s schemas/selectors-1.5.json -d tests/1.5/invalid/selectors.yml --invalid
+
+      - name: Validate dbt_project.yml sample files for dbt version 1.6
+        run: |
+          npx ajv-cli test -s schemas/dbt_project-1.6.json -d tests/1.6/valid/dbt_project.yml --valid
+          npx ajv-cli test -s schemas/dbt_project-1.6.json -d tests/1.6/invalid/dbt_project.yml --invalid
+      - name: Validate resource yml sample files for dbt version 1.6
         run: |
-          npx ajv-cli test -s schemas/dbt_yml_files.json -d tests/valid/schema.yml --valid
-          npx ajv-cli test -s schemas/dbt_yml_files.json -d tests/invalid/schema.yml --invalid
-      - name: Validate packages.yml sample files
+          npx ajv-cli test -s schemas/dbt_yml_files-1.6.json -d tests/1.6/valid/schema.yml --valid
+          npx ajv-cli test -s schemas/dbt_yml_files-1.6.json -d tests/1.6/invalid/schema.yml --invalid
+      - name: Validate packages.yml sample files for dbt version 1.6
         run: |
-          npx ajv-cli test -s schemas/packages.json -d tests/valid/packages.yml --valid
-          npx ajv-cli test -s schemas/packages.json -d tests/invalid/packages.yml --invalid
-      - name: Validate selectors.yml sample files
+          npx ajv-cli test -s schemas/packages-1.6.json -d tests/1.6/valid/packages.yml --valid
+          npx ajv-cli test -s schemas/packages-1.6.json -d tests/1.6/invalid/packages.yml --invalid
+      - name: Validate selectors.yml sample files for dbt version 1.6
+        run: |
+          npx ajv-cli test -s schemas/selectors-1.6.json -d tests/1.6/valid/selectors.yml --valid
+          npx ajv-cli test -s schemas/selectors-1.6.json -d tests/1.6/invalid/selectors.yml --invalid
```
Two thoughts: 
- A single-dimension matrix might dry this up a bit: https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs
- Is it worth testing that
  - a valid 1.6 file is rejected by the 1.5 validator, and
  -  a valid 1.5 file is still accepted by the 1.6 validator? (if we only do one of these two, it should be this one; they should all be backwards compatible, but they're not guaranteed to be forwards-incompatible)

joellabes commented on tests/validate_plusless_properties.py:
```diff
@@ -1,6 +1,12 @@
 import json 
 from jsondiff import diff
 
+PROJECT_SCHEMA_FILES = [
+    "./schemas/dbt_project.json",
+    "./schemas/dbt_project-1.5.json",
+    "./schemas/dbt_project-1.6.json"
+]
+
```
This might be better as a query that lists all files matching the pattern `dbt_project[\-\d\.]*\.json` or something, instead of a hardcoded list that we have to remember to keep up to date

joellabes commented on schemas/dependencies-1.6.json:
```diff
@@ -0,0 +1,22 @@
+{
```
lgtm

joellabes commented on tests/1.6/invalid/dependencies.yml:
```diff
@@ -0,0 +1,2 @@
+projects:
+  - gnome: my_dbt_project
```
https://www.youtube.com/watch?v=6n3pFFPSlW4

joellabes commented on tests/1.6/valid/dependencies.yml:
```diff
@@ -0,0 +1,2 @@
+projects:
+  - name: my_dbt_project
```
👍

joellabes reviewed (approved):
The ones I commented on are the only ones I looked at, assuming that the rest are renames and copy-pastes. If there's others that you'd like me to check please lmk! 

I'm approving this since the suggested changes aren't mandatory but will make life easier in the future 🔮

dave-connors-3 commented:
@joellabes would love a quick re-review! I updated the folder paths per Ben's recommendation and added the matrix feature to make CI a lot DRYer. The only notable byproduct of that move was that `schema.yml` files in the tests directory are now named `dbt_yml_files` to match the name of the schema, so the matrix combos behave properly. Let me know if that is a dealbreaker for you in any way!

I also chose _not_ to do the file regexing in `validate_plusless_properties.py` -- feels like a quarterly update of that variable as a part of the new version process is an OK amount of friction

joellabes reviewed (approved):
Magnificent!

dave-connors-3 commented on .github/workflows/ci.yml:
```diff
@@ -25,44 +32,11 @@ jobs:
-          npx ajv-cli test -s schemas/dependencies-1.6.json -d tests/1.6/invalid/dependencies.yml --invalid
+          npx ajv-cli test -s schemas/${{ matrix.dbt-version }}/${{ matrix.dbt-file-name }}-${{ matrix.dbt-version }}.json -d tests/${{ matrix.dbt-version }}/valid/${{ matrix.dbt-file-name }}.yml --valid
+          npx ajv-cli test -s schemas/${{ matrix.dbt-version }}/${{ matrix.dbt-file-name }}-${{ matrix.dbt-version }}.json -d tests/${{ matrix.dbt-version }}/invalid/${{ matrix.dbt-file-name }}.yml --invalid
+      
```
:shreks-kick:

dave-connors-3 commented:
@joellabes i didn't mean to dismiss your stale review :( i moved the default files back to the root of the folder so the IDE validation doesn't break
